packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/KotlinInteropModuleRegistry.kt

```diff
@@ -42,25 +43,33 @@ class KotlinInteropModuleRegistry(
   fun exportViewManagers(): List<ViewManager<*, *>> =
     trace("KotlinInteropModuleRegistry.exportViewManagers") {
       registry
-        .filter { it.definition.viewManagerDefinition != null }
-        .map {
-          val wrapperDelegate = ViewManagerWrapperDelegate(it)
-          when (it.definition.viewManagerDefinition!!.getViewManagerType()) {
-            ViewManagerType.SIMPLE -> SimpleViewManagerWrapper(wrapperDelegate)
-            ViewManagerType.GROUP -> GroupViewManagerWrapper(wrapperDelegate)
+        .flatMap { module ->
+          module.definition.viewManagerDefinitions.map {
+            val wrapperDelegate = ViewManagerWrapperDelegate(module, it.value, if(it.key == DEFAULT_MODULE_VIEW) module.name else null)
+            when (it.value.getViewManagerType()) {
+              ViewManagerType.SIMPLE -> SimpleViewManagerWrapper(wrapperDelegate)
+              ViewManagerType.GROUP -> GroupViewManagerWrapper(wrapperDelegate)
+            }
           }
```

Comment:
```suggestion
        .flatMap { module ->
          module.definition.viewManagerDefinitions.map { (name, definition) ->
            val wrapperDelegate = ViewManagerWrapperDelegate(module, definition, if(name == DEFAULT_MODULE_VIEW) module.name else null)
            when (definition.getViewManagerType()) {
              ViewManagerType.SIMPLE -> SimpleViewManagerWrapper(wrapperDelegate)
              ViewManagerType.GROUP -> GroupViewManagerWrapper(wrapperDelegate)
            }
          }
```

Reply:
lol, forgot that kotlin has pair destructuring <3